src/main/scala/org/apache/spark/sql/delta/OptimisticTransaction.scala

```diff
@@ -451,4 +456,29 @@ trait OptimisticTransactionImpl extends TransactionalWrite {
     logInfo(s"No logical conflicts with deltas [$checkVersion, $nextAttempt), retrying.")
     doCommit(nextAttempt, actions, attemptNumber + 1)
   }
+
+  private def isConcurrentPartitionWriteEnabled = {
+    DeltaConfigs.enableConcurrentPartitionsWrite(spark.sessionState.conf)
+  }
+
+  private def checkConcurrentWriteByPartitions(
+      myActions: Seq[Action],
+      theirActions: Seq[Action],
+      commitInfo: Option[CommitInfo]): Unit = {
+
+    val partitionMapper: Action => Iterable[String] = {
+      case f: AddFile => f.partitionValues.values
+      case _ => None
+    }
+    val theirModifiedPartitions = theirActions.flatMap(partitionMapper)
+    val myModifiedPartitions = myActions.flatMap(partitionMapper)
+
+    if (myModifiedPartitions.isEmpty || theirModifiedPartitions.isEmpty) {
+      throw new ConcurrentWriteException(commitInfo)
+    }
```

Comment:
Why is this an exception if one or both is empty?

Reply:
Does not make much sense, I'll remove this check.